nibabel/nifti2.py

```diff
@@ -221,6 +222,17 @@ def _chk_eol_check(hdr, fix=False):
             rep.fix_msg = 'setting EOL check to 13, 10, 26, 10'
         return hdr, rep
 
+    @classmethod
+    def is_header(klass, binaryblock):
+        if len(binaryblock) < 540:
```

Comment:
Nifti2 `.hdr` files are 540 bytes. Analyze/Nifti1 `.hdr` files are 348 bytes. If we end up testing a Nifti1 header with Nifti2's  `is_header`, we either need to test with `hdr = np.ndarray(shape=(), dtype=nifti1.header_dtype, buffer=binaryblock[:348])` or we need to assert that there be at least 540 bytes.

In the old style, `load` called `which_analyze_type`, which used `nifti1.header_dtype`.